### ISSUE TITLE
Minio killed by OOM-killer during large uploads

### ISSUE THREAD

**Issue by brandsr** [minio/minio#10052]
During uploads of large files minio is killed by OOM-killer. And again after first startup.

## Expected Behavior
Minio should not get into memory problems during large file uploads.

## Current Behavior
The batch (out of our scope) contains large files starts at around 02:00AM. This results in excessive memory consumption of minio and eventually being killed by the OOM-killer.

Other observations:
- large amount of fragments in `tmp` subdir:
```
[root@host tmp]# ls -1 | wc -l
118
[root@host tmp]# du -h | grep -v ^0
15G	./9619e573-5047-4e9b-8337-dc8eedbb31a9
12G	./605f3f1e-4663-4988-af82-d6b9d5c1092e
11G	./137731fa-fc57-47bf-b902-00ddbfe674d4
1.6G	./c6d54bd4-91c4-4594-8c4f-7010695b19ea
129M	./94042c44-5aaa-49db-88ba-d762f89e9154
3.0G	./b6af9046-49df-4c79-857b-8e1474b46361
1.6G	./81897ea0-2599-4d8e-8514-b771ccf88548
4.9G	./6715a6ea-a4f3-4fca-8777-16d502aab0fe
3.6G	./31ce6bcf-a435-45cf-b42f-103cb40d65d8
6.3G	./33533940-2500-4200-9377-99e5d90baa03
12G	./3ff1a4f9-7b3e-443b-acac-776a5c068a1a
4.0G	./1744ead1-afb6-4652-929f-fb0121559b8c
9.7G	./2754c7e9-6fdb-48b3-a93b-554c244b9220
9.1G	./117a4039-dcef-4445-829e-8105f11fb3c0
5.9G	./032b1e5f-3aad-4020-99e3-6a186c574343
4.8G	./f0eafd79-cdc6-4b2f-bc44-98a325c20029
103G	.
```
[minio memory issue.pdf](https://github.com/minio/minio/files/4924511/minio.memory.issue.pdf)

This while `mc ls --incomplete` does not return any fragments (empty output).

- Strange startup threads and memory consumption at startup (after being killed by OOM-killer). See attached graphs.

## Steps to Reproduce (for bugs)
N/A

## Context
Large file transfers are being killed with impact for the application.

## Your Environment
* Version used (`minio --version`): RELEASE.2020-07-02T00-15-09Z
* Server setup and configuration: 4cpu 8gb
* Operating System and version (`uname -a`): Red Hat Enterprise Linux Server release 7.2 (kernel 3.10.0-327.10.1.el7.x86_64)

**Comment by brandsr** [comment]
Hi,

as an addition a profile was created. The resulting files are attached.
[minio profile.zip](https://github.com/minio/minio/files/4924527/minio.profile.zip)

**Comment by benitoprijs** [comment]
The profile was created with the following options:
`mc admin profile start --type=cpu,mem myminio `

**Comment by klauspost** [comment]
Thanks for the report.

This should already be fixed with #9992 released in `RELEASE.2020-07-11T06-07-16Z`.

**Comment by klauspost** [comment]
All allocations in the profile originate from ListObjects calls and doesn't seem related to uploads.

Adding more memory would also be a general recommendation.
